### PR TITLE
Restructuring data kws docs

### LINKED ISSUE #1354: Documentation on data specification should clearly describe the actual current specification
Currently we have

https://docs.nnpdf.science/vp/dataspecification.html?highlight=t0#

which makes historical sense as it was written to specify the transition to the "data keyword".  But we lack a resource that focuses from the start on how to specify datasets, details on covmats filters (i.e. links to the existing page) and so on, a bit along the lines of what we have in the paper.

At minimum we should move things around so the first section concerns the current data.

### PR DESCRIPTION
**Pull request by siranipour**
I spotted a couple of formatting typos, which have been fixed. I also moved the at a glance section to the end of the document such that it reads after the section on backwards compatibility. Closes #1354

### CODE REVIEW

File: doc/sphinx/source/vp/dataspecification.rst [hunk 1]
```diff
@@ -3,80 +3,21 @@
...
-
-
 .. _datasetspec-core-label:
 
 ``DataSetSpec`` - Core dataset object
 -------------------------------------
 
+The declaration of dataset specifications within the ``validphys`` framework
+is handled using the ``dataset_input`` (or ``dataset_inputs`` namespace
+list for a collection of datasets) key. Through this keyword the user
```
Comment (RoyStegeman) [inline doc/sphinx/source/vp/dataspecification.rst:13]:
```suggestion
is handled using the ``dataset_input``  key, or ``dataset_inputs`` namespace
list for a collection of datasets. Through this keyword the user
```

File: doc/sphinx/source/vp/dataspecification.rst [hunk 2]
```diff
@@ -483,3 +425,65 @@ always require ``experiments`` to be specified in the runcard.
   group will contain all of the datasets from the fit - which is incorrect.
   This is regarded as a bug, the relevant issue is:
   https://github.com/NNPDF/reportengine/issues/38
+
+At a glance: what do I need to change in my runcards?
```
Comment (RoyStegeman) [inline doc/sphinx/source/vp/dataspecification.rst:429]:
```suggestion
What do I need to change in my runcards?
```

**Review by RoyStegeman — APPROVED** [review]
This looks good!

**Comment by siranipour** [comment]
Cheers @RoyStegeman! Will merge into master once we figure out the CI tests

**Comment by RoyStegeman** [comment]
@siranipour I think this can be merged now, right?

**Pull request merged**